### PR TITLE
Add script for travis CI test, Fixes #69

### LINKED ISSUE #69: Add proper tests for CI
It should cd to a test crate directory, that contains a single function which panics on an input of size 6 or something, then fuzz init, edit the fuzz target to call this function, run the fuzz target, and ensure a crashfile is produced of the correct length.

### PR DESCRIPTION
**Pull request by ner0x652**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 os:
   - linux
 
 matrix:
   fast_finish: true
+
 notifications:
   email: false
+
+deploy:
```
Comment (frewsxcv) [inline .travis.yml]: Can we just make this:

```
script:
  scripts/deploy.sh
```

and maybe just rename `scripts/deploy.sh` to `test/run.sh`?
  
```

Comment (frewsxcv) [inline .travis.yml]: Could also put `testcrate` in the `test/` directory

Comment (ner0x652) [inline .travis.yml]: Rename `etc` to `test`, right?

Comment (ner0x652) [inline .travis.yml]: Ok, `run` is more appropriate.

**Comment by ner0x652** [comment]
@frewsxcv @Manishearth I've renamed the directories and the `bash` script. Please review.

File: .travis.yml (hunk 1 above)
Comment (Manishearth) [inline .travis.yml]: Yeah, this has nothing to do with deploying, `script` is the way to go.

File: test/testcrate/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[package]
+name = "testcrate"
+version = "0.1.0"
+authors = ["The rust-fuzz Project Developers"]
+
```
Comment (Manishearth) [inline test/testcrate/Cargo.toml]: `publish=false`

File: .travis.yml (hunk 1 above)
Comment (Manishearth) [inline .travis.yml]: There should be a `cargo build` before the script is run, too

File: test/run.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+cd testcrate
```
Comment (Manishearth) [inline test/run.sh]: needs to cargo install before cding

Comment (ner0x652) [inline test/run.sh]: What it should install?

Comment (Manishearth) [inline test/run.sh]: cargo-fuzz. Just `cargo install` in the cargo-fuzz root dir is enough.

**Comment by ner0x652** [comment]
@Manishearth Updated.

File: .travis.yml (hunk 1 above)
Comment (frewsxcv) [inline .travis.yml]: There shouldn't be a `deploy` section here, it should just be `script`. Here's some more information:

https://docs.travis-ci.com/user/customizing-the-build#Customizing-the-Build-Step

File: test/run.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+cd ..
```
Comment (Manishearth) [inline test/run.sh]: As far as I can tell the script is run from the root anyway. Please check the travis CI results.

**Comment by Manishearth** [comment]
You need to add the bin directory to the PATH (`~/.cargo/bin` I think, but could be wrong)

Current one is causing failures on travis:

```
test/run.sh: line 4: cargo-fuzz: command not found

sed: can't read fuzz/fuzzers/fuzzer_script_1.rs: No such file or directory

test/run.sh: line 6: cargo-fuzz: command not found
```

also, this will only work on nightly, you should probably restrict this to that platform

**Comment by ner0x652** [comment]
@Manishearth I did add the path to binaries to PATH. Now there is another problem, seems related to this https://github.com/rust-lang/cargo/issues/3192

**Comment by Manishearth** [comment]
Add

```toml
[workspace]
members = ["."]
```

to the testcrate

**Comment by ner0x652** [comment]
@Manishearth unfortunately that doesn't solve the problem
```
error: multiple workspace roots found in the same workspace:
  /home/travis/build/rust-fuzz/cargo-fuzz/test/testcrate/fuzz
  /home/travis/build/rust-fuzz/cargo-fuzz/test/testcrate
```

**Comment by Manishearth** [comment]
Add a similar key to cargo-fuzz itself?

**Comment by ner0x652** [comment]
@Manishearth no, every Cargo.toml has a workspace member, but the issue remains.

**Comment by Manishearth** [comment]
I guess a workaround would be to copy this folder to the parent directory, cd there, and run the tests from there.

**Comment by ner0x652** [comment]
Seems that moving the `tescrate` to root of `cargo-fuzz` still doesn't solve the issue. Probably we should wait for a solution from Cargo team. :thinking:

**Comment by neosilky** [comment]
Can this get revisited? The issue posted to Cargo has been closed with the consensus that nightly works okay, which is fine for us.

**Comment by Manishearth** [comment]
triggered a rebuild, let's see if it works

**Comment by Manishearth** [comment]
Moved to  https://github.com/rust-fuzz/cargo-fuzz/pull/92

**Pull request closed without merging**